R/rgenoud_metaepoch.R

```diff
@@ -0,0 +1,46 @@
+#' Function that runs one GA metaepoch.
+#'
+#' @param config_rgenoud - list of rgenoud::genoud params
+#'
+#' @return list with named fields: solution, population, value
+#' @export
+#'
+#' @examples
+rgenoud_metaepoch <- function(config_rgenoud) {
+  function(fitness, suggestions, lower, upper, tree_level) {
+    config <- config_rgenoud[[tree_level]]
+    legal_passed_param_names <- Filter(function(name) {
+      name %in% methods::formalArgs(rgenoud::genoud)
+    }, names(config))
+    params <- list("pop.size" = nrow(suggestions),
+                   "max.generations" = 5)
+    for (param_name in legal_passed_param_names) {
+      params[param_name] <- config[param_name]
+    }
+    params$boundary.enforcement <- 2
+    params$hard.generation.limit <- TRUE
+    params$fn <- fitness
+    params$Domains <- matrix(c(lower, upper), ncol=2)
+    params$starting.values <- suggestions
+    params$max <- TRUE
+    params$nvars <- length(lower)
+    params$print.level <- 0
+    result <- suppressWarnings(do.call(rgenoud::genoud, params))
+    population <- rnorm_population(
```

Comment:
Patrzyłem do tej dokumentacji i też nie widzę jak wyciągnąć z tej wynikowej listy populację. No i ciężko mi powiedzieć na ile sensowne jest takie losowanie populacji. Intuicyjnie brzmi jakby mogło być kiepskie. To jest w zasadzie tak, jakby na koniec wszystko podmienić na mutacje najlepszego osobnika - a my tą szansę na mutację ustawiamy w okolicach 20-40% zazwyczaj.

 Do tego całkiem możliwe, że ten pakiet tak jak piszesz sam odpala coś nieewolucyjnego w trakcie działania jako jeden z operatorów, nie widziałem też żadnego parametru, który by to ograniczał... No ale nie wiem jak inaczej wytłumaczyć to czemu on daje takie dobre wyniki.